framework/test/gtest/BundleTrackerMethodTest.cpp

```diff
@@ -0,0 +1,480 @@
+/*=============================================================================
+
+ Library: CppMicroServices
+
+ Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+ file at the top-level directory of this distribution and at
+ https://github.com/CppMicroServices/CppMicroServices/COPYRIGHT .
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+
+ =============================================================================*/
+#include "cppmicroservices/Bundle.h"
+#include "cppmicroservices/BundleTracker.h"
+#include "cppmicroservices/Framework.h"
+#include "cppmicroservices/FrameworkEvent.h"
+#include "cppmicroservices/FrameworkFactory.h"
+
+#include "TestUtils.h"
+#include "gtest/gtest.h"
+
+#ifdef GetObject
+#  undef GetObject
+#endif
+
+using namespace cppmicroservices;
+
+class BundleTrackerMethodTest : public ::testing::Test
+{
+protected:
+  Framework framework;
+  BundleContext context;
+  BundleTracker<>::BundleStateMaskType all_states =
+    BundleTracker<>::CreateStateMask(Bundle::State::STATE_ACTIVE,
+                                     Bundle::State::STATE_INSTALLED,
+                                     Bundle::State::STATE_RESOLVED,
+                                     Bundle::State::STATE_STARTING,
+                                     Bundle::State::STATE_STOPPING,
+                                     Bundle::State::STATE_UNINSTALLED);
+
+public:
+  BundleTrackerMethodTest()
+    : framework(FrameworkFactory().NewFramework()){};
+
+  ~BundleTrackerMethodTest() override = default;
+
+  void SetUp() override
+  {
+    framework.Start();
+    context = framework.GetBundleContext();
+  }
+
+  void TearDown() override
+  {
+    framework.Stop();
+    framework.WaitForStop(std::chrono::milliseconds::zero());
+  }
+};
+
+TEST_F(BundleTrackerMethodTest, CanCreateTracker)
+{
+  ASSERT_NO_THROW(BundleTracker bundleTracker(context, all_states))
+    << "Creation of BundleTracker failed";
+}
+
+TEST_F(BundleTrackerMethodTest, GetBundlesWorks)
```

Comment:
minor; rename testpoint name - remove "Works" to be more consistent with other test point names

Reply:
Done